pkg/util/pki/generate.go

```diff
@@ -122,11 +146,18 @@ func EncodePrivateKey(pk crypto.PrivateKey, keyEncoding v1.PrivateKeyEncoding) (
 			return EncodePKCS1PrivateKey(k), nil
 		case *ecdsa.PrivateKey:
 			return EncodeECPrivateKey(k)
+		case ed25519.PrivateKey:
+			return EncodePKCS8PrivateKey(k)
```

Comment:
note for future reviewers: some useful information [here](https://crypto.stackexchange.com/questions/59850/ed25519-pkcs8-private-key-example-from-ietf-draft-seems-malformed) on this.